app/src/views/search-engine/components/search-bar/index.js

```diff
@@ -25,11 +25,23 @@ const SearchBar = ({ setSearch, refetch }) => {
               root: classes.inputRoot,
               input: classes.inputInput
             }}
-            onChange={({ target: { value } }) => setSearch(value)}
+            onChange={({ target: { value } }) => {
+              if (value?.length !== 0) {
+                setLoading(true)
+                setSearch(value)
+              } else {
+                setLoading(false)
+              }
+            }}
```

Comment:
You're probably not gonna like this question but ... how hard would it be to unit test this?

Reply:
😆 

I mean, it's totally possible. I don't think it would be "hard" per se, but of course I'd have to research a little bit to understand how to do it. Rather than hard, there is an implicit learning curve in there that naturally will take time 👁️ .

I was hoping to focus on the caching path expansion, but I could definitely prioritize this if needed.

Wdyt?

Reply:
It would be good to have, but I don't think it's more important than other improvements you could make. I'll leave it up to you to whether you want to figure out how to unit test this now or come back to it later if there's time.

Reply:
Yes, that makes sense, I think I'd like to focus on getting between today/tomorrow the caching service running, and once that gets done (and I wouldn't expect it to be THAT complex), I will definitely focus on getting the React components tested.